aws-cpp-sdk-transfer/include/aws/transfer/UploadFileRequest.h

```diff
@@ -126,6 +126,11 @@ class AWS_TRANSFER_API UploadFileRequest : public S3FileRequest, public std::ena
     bool HasPassedGetObject() const { return m_getObjectPassed.load(); }
     bool HasPassedListObjects() const { return m_listObjectsPassed.load(); }
 
+    // A map of metadata to store with the object in S3.
```

Comment:
Two overloads should work. Leave the original, then add one with the map as
const ref, then another as an r-value.
On Jan 12, 2016 10:43 AM, "kahkeng" <notifications@github.com> wrote:

> In aws-cpp-sdk-transfer/include/aws/transfer/UploadFileRequest.h
> <https://github.com/awslabs/aws-sdk-cpp/pull/76#discussion_r49495151>:
>
> > @@ -126,6 +126,11 @@ class AWS_TRANSFER_API UploadFileRequest : public S3FileRequest, public std::ena
> >      bool HasPassedGetObject() const { return m_getObjectPassed.load(); }
> >      bool HasPassedListObjects() const { return m_listObjectsPassed.load(); }
> >
> > +    // A map of metadata to store with the object in S3.
>
> Ok that makes sense, thanks for the review. I was wondering if I should go
> that route but there are many other request properties available in a
> PutObjectRequest that are yet to be implemented here (cache control,
> content encoding, request payer, etc) and this would be an api breaking
> change each time unless we did function overloading for UploadFile. Would
> that be acceptable or we're ok changing the function signature each time to
> add a new property?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/awslabs/aws-sdk-cpp/pull/76/files#r49495151>.
>